src/components/carForm.tsx

```diff
@@ -1,90 +1,106 @@
-import { ChangeEvent, FormEvent, useState } from 'react'
-import { Car } from '../types'
-import './carForm.css'
+import { ChangeEvent, FormEvent, useEffect, useState } from 'react'
+import { Car, FormErrors, emptyCar } from '../types'
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import TextField from '@mui/material/TextField';
+import { Button, ButtonGroup, FormControl, FormControlLabel, FormHelperText, FormLabel, MenuItem, Radio, RadioGroup, Typography } from '@mui/material';
 
-interface CarFormProps { cars: Car[], setCars: (cars: Car[]) => void }
-interface FormErrors extends Car {}
+import { validateCar } from '../utils'
 
-function capitalizeFirstLetter(string: string) { // Function to capitalize the first letter of a string
-    return string.charAt(0).toUpperCase() + string.slice(1)
-}
+interface CarFormProps { addCar: (car: Car) => void }
 
-export default function CarForm({ cars, setCars} : CarFormProps){
-    const [car, setCar] = useState({ maker: '', model: '', year: '' }) // Car object state
-    const [errors, setErrors] = useState<FormErrors>({
-        model: '', year: '', maker: ''
-    }) // Errors state
+export default function CarForm({ addCar } : CarFormProps){
+    const [car, setCar] = useState(emptyCar) // Car object state
+    const [errors, setErrors] = useState<FormErrors>(emptyCar) // Errors state
 
-    const validate = () => { // Function to validate form fields
-        let correct = true
-        Object.keys(errors).forEach((key) => {
-            if (!car[key as keyof Car]) {
-                errors[key as keyof Car] = `${capitalizeFirstLetter(key)} is required`
-                correct = false
-            } else {
-                errors[key as keyof Car] = ''
-            }
-        })
-        setErrors({ ...errors })
-        return correct
-    }
+    useEffect(() => { // Effect to re-validate form fields
+        if (Object.values(errors).some((value) => value !== '')){
+            validateCar({car, errors, setErrors})
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [car])
```

Comment:
Great idea to add this `useEffect` to validate the form onChange. 
In the next blocks we will discuss some packages that allow to do this automatically